File: java/arrays_hashing/two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int len = nums.length;
        int[] res = new int[2];
        HashMap<Integer, Integer> hash = new HashMap<>();

        for(int i=0; i<len; i++){
            int cur = target - nums[i];

            if(hash.get(cur) != null){
                res[0] = hash.get(cur);
                res[1] = i;
                return res;
            }
            else{
                hash.put(nums[i],i);
            }
        }
        return res;
    }
}

File: python/arrays_hashing/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}
        for index, num in enumerate(nums):
            if num in hm:
                return [index, hm[num]]
            else:
                hm[target-num] = index
        return []

        